scientific-computing-with-python/learn-lambda-functions-by-building-a-expense-tracker/expense-tracker.py

```diff
@@ -0,0 +1,54 @@
+expenses = []
+
+def add_expense(amount, category):
+    expenses.append({"amount": amount, "category": category})
+
+def list_expenses():
+    for expense in expenses:
+        print(f"Amount: {expense['amount']}, Category: {expense['category']}")
+
+def total_expenses():
+    return sum(map(lambda expense: expense['amount'], expenses))
+
+def filter_expenses_by_category(category):
+    filtered_expenses = filter(lambda expense: expense['category'] == category, expenses)
+    for expense in filtered_expenses:
+        print(f"Amount: {expense['amount']}, Category: {expense['category']}")
```

Comment:
```suggestion
    return filter(lambda expense: expense['category'] == category, expenses)
```
After that `filter_expenses_by_category` only filters expenses.